### PR TITLE
INTMDB-828: [Terraform] Missing Schedule Parameter in Resource mongodbatlas_online_archive

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Ticket: [INTMDB-828](https://jira.mongodb.org/browse/INTMDB-828)

This PR adds the schedule field to the `mongodbatlas_online_archive` resource and data source

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments

```bash
Running tool: /Users/andrea.angiolillo/.asdf/shims/go test -timeout 300000s -run ^TestAccBackupRSOnlineArchive$ github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas

ok  	github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas	1060.933s
```

### CODE REVIEW

**Review by martinstibbe — APPROVED** [review]
LGTM

File: mongodbatlas/resource_mongodbatlas_online_archive.go [hunk 1]
```diff
@@ -283,6 +325,13 @@ func resourceMongoDBAtlasOnlineArchiveImportState(ctx context.Context, d *schema
 		log.Printf("error setting project id %s for Online Archive id: %s", err, atlasID)
 	}
 
+	mapValues := fromOnlineArchiveToMapInCreate(outOnlineArchive)
+	for key, val := range mapValues {
+		if err := d.Set(key, val); err != nil {
+			return nil, fmt.Errorf("error MongoDB Atlas Online Archive with id %s, read error %s", atlasID, err.Error())
```
Comment (gssbzn) [inline mongodbatlas/resource_mongodbatlas_online_archive.go]:
```suggestion
			return nil, fmt.Errorf("error MongoDB Atlas Online Archive with id %s, read error: %w", atlasID, err)
```

**Review by gssbzn — APPROVED** [review]

File: website/docs/d/online_archive.html.markdown [hunk 2]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
-* `partitio_fields.field_type` - Type of the partition field
 * `state`    - Status of the online archive. Valid values are: Pending, Archiving, Idle, Pausing, Paused, Orphaned and Deleted
 
+### Criteria
+* `type`          - Type of criteria (DATE, CUSTOM)
+* `date_field`    - Name of an already indexed date field from the documents. Data is archived when the current date is greater than the value of the date field specified here plus the number of days specified via the `expire_after_days` parameter.
+* `date_format`   - the date format. Valid values:  ISODATE (default), EPOCH_SECONDS, EPOCH_MILLIS, EPOCH_NANOSECONDS
+* `expire_after_days` - Number of days that specifies the age limit for the data in the live Atlas cluster. Data is archived when the current date is greater than the value of the date field specified via the `date_field` parameter plus the number of days specified here.
+* `query` - JSON query to use to select documents for archiving. Atlas uses the specified query with the db.collection.find(query) command. The empty document {} to return all documents is not supported.
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: should we note that this is required when `type` is `CUSTOM`?

File: website/docs/d/online_archive.html.markdown [hunk 3]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
-* `partition_fields` -  Fields to use to partition data.
-* `partition_fields.field_name` - Name of the field. To specify a nested field, use the dot notation.
-* `partition_fields.order` - Position of the field in the partition. Value can be: 0,1,2
-By default, the date field specified in the criteria.dateField parameter is in the first position of the partition.
-* `partitio_fields.field_type` - Type of the partition field
 * `state`    - Status of the online archive. Valid values are: Pending, Archiving, Idle, Pausing, Paused, Orphaned and Deleted
 
+### Criteria
+* `type`          - Type of criteria (DATE, CUSTOM)
+* `date_field`    - Name of an already indexed date field from the documents. Data is archived when the current date is greater than the value of the date field specified here plus the number of days specified via the `expire_after_days` parameter.
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: should we note that `date_field`, `date_format`, and `expire_after_days` apply only when `type` is `DATE`? Not sure how conditional fields work in Terraform

Comment (jwilliams-mongo) [reply website/docs/d/online_archive.html.markdown]: we have an updated description of date_field in the API docs:

Indexed database parameter that stores the date that determines when data moves to the online archive. Atlas archives the data when the current date exceeds the date in this database parameter plus the number of days specified through the expire_after_days parameter.

File: website/docs/d/online_archive.html.markdown [hunk 4]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+* `date_format`   - the date format. Valid values:  ISODATE (default), EPOCH_SECONDS, EPOCH_MILLIS, EPOCH_NANOSECONDS
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: we have more detail on this field in the API docs:

Syntax used to write the date after which data moves to the online archive. Date can be expressed as ISO 8601 or Epoch timestamps. The Epoch timestamp can be expressed as nanoseconds, milliseconds, or seconds. Set this parameter when "criteria.type" : "DATE". You must set "criteria.type" : "DATE" if "collectionType": "TIMESERIES".

File: website/docs/d/online_archive.html.markdown [hunk 5]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+* `expire_after_days` - Number of days that specifies the age limit for the data in the live Atlas cluster. Data is archived when the current date is greater than the value of the date field specified via the `date_field` parameter plus the number of days specified here.
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: we have more detail on this in the API docs:

Number of days after the value in the criteria.dateField when MongoDB Cloud archives data in the specified cluster. Set this parameter when you set "criteria.type" : "DATE".

File: website/docs/d/online_archive.html.markdown [hunk 6]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+* `query` - JSON query to use to select documents for archiving. Atlas uses the specified query with the db.collection.find(query) command. The empty document {} to return all documents is not supported.
+
+### Schedule
+
+* `type`          - Type of schedule (`DEFAULT`, `DAILY`, `MONTHLY`, `WEEKLY`).
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: what is default here? can we also replace the tabs between the field names and the hyphens with spaces? That seems to be the convention used elsewhere on this page

File: website/docs/d/online_archive.html.markdown [hunk 7]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+* `start_hour`    - Hour of the day when the when the scheduled window to run one online archive starts.  
+* `end_hour`      - Hour of the day when the scheduled window to run one online archive ends.
+* `start_minute`   - Minute of the hour when the scheduled window to run one online archive starts.
+* `end_minute`     - Minute of the hour when the scheduled window to run one online archive ends.
+* `day_of_month`   - Day of the month when the scheduled archive starts.
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: should we note that this applies only when `type` is `MONTHLY`?

File: website/docs/d/online_archive.html.markdown [hunk 8]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+* `day_of_week`     - Day of the week when the scheduled archive starts. The week starts with Monday (1) and ends with Sunday (7).
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: should we note that this applies only when `type` is `WEEKLY`?

File: website/docs/d/online_archive.html.markdown [hunk 9]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+
+### Partition
+* `field_name` - Name of the field. To specify a nested field, use the dot notation.
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: we have more detail on this in the API docs:

Human-readable label that identifies the parameter that Atlas uses to partition data. To specify a nested parameter, use the dot notation.

File: website/docs/d/online_archive.html.markdown [hunk 10]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+* `order` - Position of the field in the partition. Value can be: 0,1,2
+By default, the date field specified in the criteria.dateField parameter is in the first position of the partition.
+* `field_type` - (Optional) type of the partition field
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: we have more detail on this in the API docs: Enum: "date" "int" "long" "objectId" "string" "uuid"
Data type of the parameter that that Atlas uses to partition data. Partition parameters of type [UUID](http://bsonspec.org/spec.html) must be of binary subtype 4. MongoDB Cloud skips partition parameters of type UUID with subtype 3.

File: website/docs/d/online_archive.html.markdown [hunk 11]
```diff
@@ -34,19 +34,31 @@ data "mongodbatlas_online_archive" "test" {
...
+* `start_hour`    - Hour of the day when the when the scheduled window to run one online archive starts.  
+* `end_hour`      - Hour of the day when the scheduled window to run one online archive ends.
+* `start_minute`   - Minute of the hour when the scheduled window to run one online archive starts.
+* `end_minute`     - Minute of the hour when the scheduled window to run one online archive ends.
+* `day_of_month`   - Day of the month when the scheduled archive starts.
+* `day_of_week`     - Day of the week when the scheduled archive starts. The week starts with Monday (1) and ends with Sunday (7).
+
+### Partition
+* `field_name` - Name of the field. To specify a nested field, use the dot notation.
+* `order` - Position of the field in the partition. Value can be: 0,1,2
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archive.html.markdown]: we have more detail on this in the API docs:

Sequence in which Atlas slices the collection data to create partitions. The resource expresses this sequence starting with zero. The value of the criteria.dateField parameter defaults as the first item in the partition sequence.

File: website/docs/d/online_archives.html.markdown [hunk 12]
```diff
@@ -34,26 +34,32 @@ In addition to all arguments above, the following attributes are exported:
...
-* `criteria.query` - JSON query to use to select documents for archiving. Only for `CUSTOM` type
-* `partition_fields` -  Fields to use to partition data.
-* `partition_fields.field_name` - Name of the field. To specify a nested field, use the dot notation.
-* `partition_fields.order` - Position of the field in the partition. Value can be: 0,1,2
-By default, the date field specified in the criteria.dateField parameter is in the first position of the partition.
-* `partitio_fields.field_type` - Type of the partition field
 * `state`    - Status of the online archive. Valid values are: Pending, Archiving, Idle, Pausing, Paused, Orphaned and Deleted
 
-See [MongoDB Atlas API](https://docs.atlas.mongodb.com/reference/api/online-archive-get-all-for-cluster/) Documentation for more information.
+### Criteria
```
Comment (jwilliams-mongo) [inline website/docs/d/online_archives.html.markdown:44]: see all comments above

File: website/docs/r/online_archive.html.markdown [hunk 13]
```diff
@@ -81,28 +88,38 @@ resource "mongodbatlas_online_archive" "test" {
 * `partition_fields` -  (Recommended) Fields to use to partition data. You can specify up to two frequently queried fields to use for partitioning data. Note that queries that don’t contain the specified fields will require a full collection scan of all archived documents, which will take longer and increase your costs. To learn more about how partition improves query performance, see [Data Structure in S3](https://docs.mongodb.com/datalake/admin/optimize-query-performance/#data-structure-in-s3). The value of a partition field can be up to a maximum of 700 characters. Documents with values exceeding 700 characters are not archived.
 * `paused`           - (Optional) State of the online archive. This is required for pausing an active or resume a paused online archive. The resume request will fail if the collection has another active online archive.
 
-### Criteria details
+### Criteria
 
 There are two types of criteria, `DATE` to select documents for archiving based on a date and
 `CUSTOM` to select documents for archiving based on a custom JSON query.
```
Comment (jwilliams-mongo) [inline website/docs/r/online_archive.html.markdown:95]: see all comments above

File: website/docs/d/online_archives.html.markdown (hunk 12 above)
Comment (Zuhairahmed) [reply website/docs/d/online_archives.html.markdown:44]: appreciate all the great feedback @jwilliams-mongo

**Review by Zuhairahmed — APPROVED** [review]
LGTM pending doc update suggestions from @jwilliams-mongo, thanks @andreaangiolillo!

File: website/docs/d/online_archive.html.markdown (hunk 6 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: `DEFAULT` seems to be one of the allowed values for type. I will double-check the endpoint in mms to make sure that is correct.
Screenshot 2023-07-06 at 08 57 21

File: website/docs/d/online_archive.html.markdown (hunk 2 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: Updated

File: website/docs/d/online_archive.html.markdown (hunk 3 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/d/online_archive.html.markdown (hunk 4 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/d/online_archive.html.markdown (hunk 5 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/d/online_archive.html.markdown (hunk 7 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/d/online_archive.html.markdown (hunk 8 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/d/online_archive.html.markdown (hunk 9 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/d/online_archive.html.markdown (hunk 10 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/d/online_archive.html.markdown (hunk 11 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: updated

File: website/docs/r/online_archive.html.markdown (hunk 13 above)
Comment (andreaangiolillo) [reply website/docs/r/online_archive.html.markdown:95]: updated

File: website/docs/d/online_archives.html.markdown (hunk 12 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archives.html.markdown:44]: updated

File: website/docs/d/online_archive.html.markdown (hunk 6 above)
Comment (andreaangiolillo) [reply website/docs/d/online_archive.html.markdown]: Looking at the mms codebase, `DEFAULT` is a valid value for type.

> can we also replace the tabs between the field names and the hyphens with spaces? That seems to be the convention used elsewhere on this page

I updated the valid values. Let me know if this format is okay. Thanks

**Comment by andreaangiolillo** [comment]
Hi @jwilliams-mongo, merging this PR as I addressed all your comments. I will open a follow-up PR to address any other comments if any. Thanks!

File: website/docs/r/online_archive.html.markdown [hunk 14]
```diff
@@ -81,28 +88,37 @@ resource "mongodbatlas_online_archive" "test" {
...
+* `date_format`   - Syntax used to write the date after which data moves to the online archive. Date can be expressed as ISO 8601 or Epoch timestamps. The Epoch timestamp can be expressed as nanoseconds, milliseconds, or seconds. You must set `type` to `DATE` if `collectionType` is `TIMESERIES`. Valid values:  ISODATE (default), EPOCH_SECONDS, EPOCH_MILLIS, EPOCH_NANOSECONDS.
+* `expire_after_days` - Number of days after the value in the criteria.dateField when MongoDB Cloud archives data in the specified cluster.
 
 The only field required for criteria type `CUSTOM`
 
-* `criteria.query` - JSON query to use to select documents for archiving. Atlas uses the specified query with the db.collection.find(query) command. The empty document {} to return all documents is not supported.
+* `query` - JSON query to use to select documents for archiving. Atlas uses the specified query with the db.collection.find(query) command. The empty document {} to return all documents is not supported.
+
+### Schedule
```
Comment (X-Guardian) [inline website/docs/r/online_archive.html.markdown:108]: `schedule` should have also been added to the `Argument Reference` section above.

**Pull request merged**